upb/def.c

```diff
@@ -2357,6 +2363,50 @@ static void create_service(
   }
 }
 
+static int popcount(uint64_t x) {
+  // For assertions only, speed does not matter.
+  int n = 0;
+  while (x) {
+    if (x & 1) n++;
+    x >>= 1;
+  }
+  return n;
+}
+
+upb_enumlayout *create_enumlayout(symtab_addctx *ctx, const upb_enumdef *e) {
+  int n = 0;
+  uint64_t mask = 0;
+
+  for (int i = 0; i < e->value_count; i++) {
+    uint32_t val = (uint32_t)e->values[i].number;
+    if (val < 64) {
+      mask |= 1 << val;
+    } else {
+      n++;
+    }
+  }
+
+  int32_t *values = symtab_alloc(ctx, sizeof(*values) * n);
+  int32_t *p = values;
+
+  for (int i = 0; i < e->value_count; i++) {
+    int32_t val = e->values[i].number;
+    if ((uint32_t)val >= 64) {
+      *p++ = val;
+    }
+  }
```

Comment:
I am simply not following what this loop is doing.

Assigning val to subsequent slots in p assuming it is > 64

Why is that useful?

Reply:
can you add a comment in the code pointing to the place where it is documented?